projects/plugins/boost/app/modules/cache/Boost_Cache.php

```diff
@@ -0,0 +1,194 @@
+<?php
+
+namespace Automattic\Jetpack_Boost\Modules\Page_Cache;
+
+abstract class Boost_Cache {
+	/*
+	 * @var string - The path key used to identify the cache directory for the current request. MD5 of the request uri.
+	 */
+	protected $path_key = false;
+
+	/*
+	 * @var string - The normalized path for the current request. This is not sanitized. Only to be used for comparison purposes.
+	 */
+	protected $request_uri = false;
+
+	public function __construct() {
+		$this->request_uri = isset( $_SERVER['REQUEST_URI'] )
+			? $this->normalize_request_uri( $_SERVER['REQUEST_URI'] ) // phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotSanitized, WordPress.Security.ValidatedSanitizedInput.MissingUnslash
+			: false;
+	}
+
+	/*
+	 * Serve the cached page if it exists, otherwise start output buffering.
+	 */
+	public function serve() {
+		if ( ! $this->get() ) {
+			$this->ob_start();
+		}
+	}
+
+	/*
+	 * Returns true if the request is cacheable.
+	 *
+	 * If a request is in the backend, or is a POST request, or is not an
+	 * html request, it is not cacheable.
+	 * The filter boost_cache_cacheable can be used to override this.
+	 *
+	 * @return bool
+	 */
+	public function is_cacheable() {
+		if ( ! apply_filters( 'boost_cache_cacheable', $this->request_uri ) ) {
+			return false;
+		}
+
+		if ( function_exists( 'is_404' ) && is_404() ) {
+			return false;
+		}
+
+		if ( function_exists( 'is_feed' ) && is_feed() ) {
+			return false;
+		}
+
+		if ( $this->is_backend() ) {
+			return false;
+		}
+
+		if ( isset( $_SERVER['REQUEST_METHOD'] ) && $_SERVER['REQUEST_METHOD'] === 'POST' ) {
```

Comment:
Shouldn't we cache only GET requests and ignore everything(PUT, PATCH, DELETE, etc.) else? I think it would be more accurate to only allow GET specifically.
```suggestion
		if ( isset( $_SERVER['REQUEST_METHOD'] ) && $_SERVER['REQUEST_METHOD'] !== 'GET' ) {
```

Reply:
Thanks, you're right. Applied the change!